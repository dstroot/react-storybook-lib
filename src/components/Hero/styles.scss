.hero {
  width: 100%;
  // padding: 60px 30px;

  /* Set a specific min height */
  height: 60vw;

  /* Position and center the image to scale nicely on all screens */
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;

  // .color-wash {
  //   /* Set a specific min height */
  //   min-height: 60vh;
  //   max-height: 80vh;
  //   /* color overlay */
  //   background-color: rgba(0, 0, 0, 0.5);
  // }
}

.bg-shade {
  background-color: rgba(0, 0, 0, 0.5);
}
