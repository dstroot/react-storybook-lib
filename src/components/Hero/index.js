import React from "react";
import PropTypes from "prop-types";

// Styles
import "./styles.scss";

const Hero = ({ align, image, tagline, text, attribution }) => {
  return (
    <div
      className="hero text-white"
      style={{
        backgroundImage: `url(${image})`,
        backgroundSize: `cover`
      }}
    >
      <div className="container-fluid">
        {align === "left" && (
          <div className="row align-items-center p-2">
            <div className="col-md-4 offset-md-1 p-lg-5 my-5 bg-shade">
              <h6 className="text-uppercase">{tagline || ""}</h6>
              <h3 className="font-weight-light">{text || ""}</h3>
              <p className="font-weight-light">{attribution || ""}</p>
            </div>
          </div>
        )}
        {align === "center" && (
          <>
            <div className="row pt-5 mt-5" />
            <div className="row pt-5 align-items-bottom">
              <div className="col-md-6 offset-md-3 mt-5 p-3 text-center bg-shade">
                <h6 className="text-uppercase">{tagline || ""}</h6>
                <h1 className="display-4">{text || ""}</h1>
                <p className="font-weight-light">{attribution || ""}</p>
              </div>
            </div>
          </>
        )}
        {align === "right" && (
          <div className="row align-items-center p-2">
            <div className="col-md-4 offset-md-7 p-lg-5 my-5 bg-shade">
              <h6 className="text-uppercase">{tagline || ""}</h6>
              <h3 className="font-weight-light">{text || ""}</h3>
              <p className="font-weight-light">{attribution || ""}</p>
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

Hero.propTypes = {
  /** Where to place the text (left/center/right) */
  align: PropTypes.string.isRequired,
  /** The background image URL */
  image: PropTypes.string.isRequired,
  /** The tagline */
  tagline: PropTypes.string,
  /** Some text */
  text: PropTypes.string,
  /** Quote attribution */
  attribution: PropTypes.string
};

export default Hero;
