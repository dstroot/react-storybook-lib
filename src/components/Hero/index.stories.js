import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import Hero from '../Hero';

// images
import image1 from './media/1.jpg';
import image2 from './media/2.jpg';
import image3 from './media/3.jpg';
import image4 from './media/startup.jpg';
import image5 from './media/home.jpg';

storiesOf('Hero', module)
  .add('show Hero image 1', () => (
    <Hero
      align="right"
      image={image1}
      text="Achieve your most important financial goals with Pacific Life"
    />
  ))
  .add('show Hero image 2', () => (
    <Hero
      align="left"
      image={image2}
      text="Achieve your most important financial goals with Pacific Life"
    />
  ))
  .add('show Hero image 3', () => (
    <Hero
      image={image3}
      align="center"
      tagline="The Power of Pacific Life"
      text="Achieve your most important financial goals with Pacific Life"
    />
  ))
  .add('show Hero image 4', () => (
    <Hero
      image={image4}
      align="right"
      tagline="The Power of Pacific Life"
      text="Achieve your most important financial goals with Pacific Life"
      attribution=" - Jim Morris"
    />
  ))
  .add('show Hero image 5', () => (
    <Hero
      image={image5}
      align="center"
      tagline="The Power of Pacific Life"
      text="Achieve your most important financial goals with Pacific Life"
    />
  ));
