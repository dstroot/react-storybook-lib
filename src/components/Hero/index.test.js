import React from 'react';
import renderer from 'react-test-renderer';

import Hero from '../Hero';
import image from '../../components/Hero/media/startup.jpg';

test('HeroSection renders some content', () => {
  const component = renderer.create(<Hero image={image} align="left" />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
