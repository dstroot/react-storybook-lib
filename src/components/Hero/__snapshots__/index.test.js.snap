// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection renders some content 1`] = `
<div
  className="hero text-white"
  style={
    Object {
      "backgroundImage": "url(startup.jpg)",
      "backgroundSize": "cover",
    }
  }
>
  <div
    className="container-fluid"
  >
    <div
      className="row align-items-center p-2"
    >
      <div
        className="col-md-4 offset-md-1 p-lg-5 my-5 bg-shade"
      >
        <h6
          className="text-uppercase"
        >
          
        </h6>
        <h3
          className="font-weight-light"
        >
          
        </h3>
        <p
          className="font-weight-light"
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;
